mm2src/coins/eth/eth_swap_v2.rs

```diff
@@ -96,15 +127,379 @@ impl EthCoin {
+            EthCoinType::Nft { .. } => {
+                return Err(TransactionErr::ProtocolNotSupported(ERRL!(
+                    "NFT protocol is not supported for ETH and ERC20 Swaps"
+                )))
+            },
+        };
+        let (taker_swap_v2_contract, approve_func, token_address) = self
+            .taker_swap_v2_details(TAKER_PAYMENT_APPROVE, TAKER_PAYMENT_APPROVE)
+            .await?;
+        let decoded = try_tx_s!(decode_contract_call(approve_func, gen_args.taker_tx.unsigned().data()));
+        let taker_status = try_tx_s!(
+            self.payment_status_v2(
+                taker_swap_v2_contract,
+                decoded[0].clone(),
+                &TAKER_SWAP_V2,
+                EthPaymentType::TakerPayments,
+                3,
+            )
+            .await
+        );
+        validate_payment_state(
+            gen_args.taker_tx,
+            taker_status,
+            TakerPaymentStateV2::TakerApproved as u8,
+        )
+        .map_err(|e| TransactionErr::Plain(ERRL!("{}", e)))?;
+        let data = try_tx_s!(
+            self.prepare_spend_taker_payment_data(gen_args, secret, decoded, token_address)
+                .await
+        );
+        let spend_payment_tx = self
+            .sign_and_send_transaction(
+                U256::from(ZERO_VALUE),
+                Action::Call(taker_swap_v2_contract),
+                data,
+                gas_limit,
+            )
+            .compat()
+            .await?;
+        Ok(spend_payment_tx)
+    }
+
+    /// Checks that taker payment state is `MakerSpent`.
+    /// Accepts maker spent payment transaction and returns it if payment status is correct.
+    pub(crate) async fn wait_for_taker_payment_spend_impl(
+        &self,
+        taker_payment: &SignedEthTx,
+        wait_until: u64,
+    ) -> MmResult<SignedEthTx, WaitForTakerPaymentSpendError> {
+        let (decoded, taker_swap_v2_contract) = self
+            .get_decoded_and_swap_contract(taker_payment, "spendTakerPayment")
+            .await?;
+        loop {
+            let taker_status = self
+                .payment_status_v2(
+                    taker_swap_v2_contract,
+                    decoded[0].clone(), // id from spendTakerPayment
+                    &TAKER_SWAP_V2,
+                    EthPaymentType::TakerPayments,
+                    3,
+                )
+                .await?;
+            if taker_status == U256::from(TakerPaymentStateV2::MakerSpent as u8) {
+                return Ok(taker_payment.clone());
+            }
+            let now = now_sec();
+            if now > wait_until {
+                return MmError::err(WaitForTakerPaymentSpendError::Timeout { wait_until, now });
+            }
+            Timer::sleep(10.).await;
+        }
+    }
+
+    /// Prepares data for EtomicSwapTakerV2 contract `ethTakerPayment` method
+    ///     function ethTakerPayment(
```

Comment:
Do we need contract parameter list here in comments? Maybe just put here a link to the contract

Reply:
https://github.com/KomodoPlatform/komodo-defi-framework/pull/2169/commits/6dd0fc296e0b2a2afcc0c377875ef9db9b86dd1a

Not so need, I just wanted devs to have params list and their types right in front of eyes. Lets add links to smart contract methods, so dev could find smart contract source code.